NuRadioReco/detector/generic_detector.py

```diff
@@ -155,40 +178,48 @@ def __init__(self, json_filename, default_station=None, default_channel=None, de
         for sta in self._stations.all():
             if "reference_station" in sta:
                 ref = self._stations.get(
-                  (Station.station_id == sta["reference_station"]))
+                    (Station.station_id == sta["reference_station"]))
                 if ref is None:
                     raise ValueError(
                         'The reference station {} was not found in the detector description'.format(
-                            ref["reference_station"]))
+                            sta["reference_station"]))
 
         Channel = Query()
         for chan in self._channels.all():
             if "reference_channel" in chan:
                 ref = self._channels.get(
-                  (Channel.station_id == chan["station_id"]) & (Channel.channel_id == chan["reference_channel"]))
+                    (Channel.station_id == chan["station_id"]) & (Channel.channel_id == chan["reference_channel"]))
                 if ref is None:
-                    raise ValueError(
-                        'The reference channel {} of station {} was not found in the detector description'.format(
-                            ref["reference_channel"], ref["station_id"]))
+                    # Check if reference channel sits in reference station
+                    reference_station_id = self.__lookuptable_reference_station[chan["station_id"]]
+                    ref = self._channels.get(
+                        (Channel.station_id == reference_station_id) &
+                        (Channel.channel_id == chan["reference_channel"])
+                    )
+                    if ref is None:
+                        raise ValueError(
+                            'The reference channel {} of station {} was not found in the detector description'.format(
+                                chan["reference_channel"], chan["station_id"]))
```

Comment:
Sorry for weighing in so late and thanks for fixing this. However, I think the previous check (l.190-191) (at least in the current implementation) is simply wrong and should be removed - right now, if the reference_channel is present in the station but not in the reference_station, this will not raise an error here, but it will raise an error when actually trying to read in the reference_channel in l.269-271, because there we always (and only) check the reference_station, not the station.

Reply:
I see, but shouldn't this be an option, to specify a reference per station, even if the station has a reference station? Cause then we should actually change the logic of lines 269-271.